### PR TITLE
only allow trusted env variables in sti builder container

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
@smarterclayton fyi, @csrwng ptal.

File: pkg/build/controller/strategy/util.go [hunk 1]
```diff
@@ -99,16 +99,31 @@ func setupDockerSecrets(pod *kapi.Pod, pushSecret string) {
...
+// will be merged.
+func mergeTrustedEnvWithoutDuplicates(source []kapi.EnvVar, output *[]kapi.EnvVar) {
+
+	// filter out all environment variables except trusted/well known
+	// values, because we do not want random environment variables being
+	// fed into the privileged STI container via the BuildConfig definition.
+	filteredSource := []kapi.EnvVar{}
+	for _, env := range source {
+		switch env.Name {
+		case "BUILD_LOGLEVEL":
```
Comment (mfojtik) [inline pkg/build/controller/strategy/util.go]: can you make this a global list variable?

var blacklistEnvVarNames=string[]{"BUILD_LOGLEVEL"}

Comment (bparees) [inline pkg/build/controller/strategy/util.go]: it's not a blacklist, it's a whitelist, but sure.

**Comment by bparees** [comment]
@mfojtik updated.

**Comment by bparees** [comment]
(though frankly it's quite a bit uglier now, until we actually have multiple trusted env variables it's a lot of code for a simple check :) )

**Comment by smarterclayton** [comment]
HTTP_PROXY and http_proxy?

----- Original Message -----

**Comment by smarterclayton** [comment]
Although wait... no, we don't want that.  Scratch that.

----- Original Message -----

**Comment by bparees** [comment]
@csrwng anything else? i have an itchy merge finger today.

**Comment by csrwng** [comment]
@bparees nope. LGTM [merge]

**Comment by bparees** [comment]
[merge]

**Comment by bparees** [comment]
fixes https://github.com/openshift/origin/issues/1937

**Pull request merged**